src/layers/attention/self_attention.rs

```diff
@@ -312,6 +312,9 @@ pub enum SelfAttentionError {
         n_query_heads: usize,
     },
 
+    #[snafu(display("Cannot concatenate append key and value to cache"))]
+    KeyValueCache { source: candle_core::Error },
```

Comment:
`ConcatKVCache` instead as the other variants describe actions (not consistently, though)?

Reply:
Renamed.